database/skylink.go

```diff
@@ -72,34 +82,83 @@ func (db *DB) SkylinkFetch(ctx context.Context, sl string) (Skylink, error) {
 		return Skylink{}, sr.Err()
 	}
 	s := Skylink{}
-	err := sr.Decode(&s)
+	err = sr.Decode(&s)
 	if err != nil {
 		return Skylink{}, err
 	}
 	return s, nil
 }
 
+// SkylinkMarkPinned marks a skylink as pinned (or no longer unpinned), meaning
+// that Pinner should make sure it's pinned by the minimum number of servers.
+func (db *DB) SkylinkMarkPinned(ctx context.Context, skylink string) error {
+	sl, err := CanonicalSkylink(skylink)
+	if err != nil {
+		return ErrInvalidSkylink
+	}
+	filter := bson.M{"skylink": sl}
+	update := bson.M{"$set": bson.M{"unpin": false}}
+	opts := options.UpdateOptions{}
+	opts.SetUpsert(true)
+	_, err = db.staticDB.Collection(collSkylinks).UpdateOne(ctx, filter, update)
+	return err
+}
+
+// SkylinkMarkUnpinned marks a skylink as unpinned, meaning that all servers
+// should stop pinning it.
+func (db *DB) SkylinkMarkUnpinned(ctx context.Context, skylink string) error {
+	sl, err := CanonicalSkylink(skylink)
+	if err != nil {
+		return ErrInvalidSkylink
+	}
+	filter := bson.M{"skylink": sl}
+	update := bson.M{"$set": bson.M{"unpin": true}}
+	opts := options.UpdateOptions{}
+	opts.SetUpsert(true)
```

Comment:
this is probably ok but just checking in case this is  a copy-paste mistake, do you want to upsert on unpinning also?

Reply:
Yes. The reason being that we mark a skylink as unpinned when we actively want to stop pinning in on all of our servers. The motivation for that would be that no user is currently pinning that skylink. So, if the skylink doesn't exist in our DB we want to create it in order to notify all servers that they need to remove it.